File: app/src/main/java/ru/shprot/sudokumobdevkz/model/database/StatisticDao.java
package ru.shprot.sudokumobdevkz.model.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import ru.shprot.sudokumobdevkz.model.game.Statistic;

@Dao
public interface StatisticDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertStatistic(Statistic statistic);

    @Query("SELECT * FROM statistic_table WHERE difficulty = :difficulty")
    Statistic getStatistic(int difficulty);

    @Query("DELETE FROM statistic_table WHERE difficulty = :difficulty")
    void removeCurrentStatistic(int difficulty);
}

File: app/src/main/java/ru/shprot/sudokumobdevkz/core/base/data/database/dao/StatisticDao.kt
package ru.shprot.sudokumobdevkz.core.base.data.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import ru.shprot.sudokumobdevkz.core.base.data.database.entity.StatisticEntity

@Dao
interface StatisticDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsert(statistic: StatisticEntity)

    @Query("SELECT * FROM statistic_table WHERE difficulty = :difficulty")
    suspend fun getByDifficulty(difficulty: Int): StatisticEntity?

    @Query("SELECT * FROM statistic_table WHERE difficulty = :difficulty")
    fun observeByDifficulty(difficulty: Int): Flow<StatisticEntity?>

    @Query("SELECT * FROM statistic_table")
    fun observeAll(): Flow<List<StatisticEntity>>

    @Query("SELECT * FROM statistic_table")
    suspend fun getAll(): List<StatisticEntity>

    @Query("DELETE FROM statistic_table WHERE difficulty = :difficulty")
    suspend fun deleteByDifficulty(difficulty: Int)
}
